sktime/datatypes/_vectorize.py

```diff
@@ -271,10 +271,14 @@ def _iter_cols(inst, group_name=None):
 
         iter_levels = self._iter_levels(iterate_as)
         is_self_iter = len(iter_levels) == self.X_mi_index.nlevels
+        if len(iter_levels) == 1:  #
+            iter_levels = iter_levels[0]
 
         if is_self_iter:
             yield from _iter_cols(self.X_multiindex)
         else:
+            if len(iter_levels) == 1:  # single level, groupby expects scalar
```

Comment:
From this failure, this line seems problematic, at least in some cases.

https://github.com/sktime/sktime/actions/runs/7583400865/job/20657552098#step:5:238

Reply:
yes, copy-paste error or merge error, the fix was pasted twice, so first time it would unpack and second time it would fail.

Fixed now, with additional safety per type check.